diff --git a~/source3/registry/reg_parse_internal.h a/source3/registry/reg_parse_internal.h
index 14fed89..ebf1cbc 100644
--- a~/source3/registry/reg_parse_internal.h
+++ a/source3/registry/reg_parse_internal.h
@@ -37,8 +37,7 @@ struct cbuf;
 #define USE_NATIVE_ICONV
 #if defined USE_NATIVE_ICONV && defined HAVE_NATIVE_ICONV
 #  define smb_iconv_t     iconv_t
-#  define smb_iconv(CD, IPTR, ILEN, OPTR, OLEN) \
-	iconv(CD, (char**)(IPTR), ILEN, OPTR, OLEN)
+#  define smb_iconv(CD, IPTR, ILEN, OPTR, OLEN) iconv(CD, (const char **)IPTR, ILEN, OPTR, OLEN)
 #  define smb_iconv_open  iconv_open
 #  define smb_iconv_close iconv_close
 #endif
